### ISSUE TITLE
false positive in OrphanedFormatString check

### ISSUE THREAD

**Issue by msridhar** [google/error-prone#1021]
### What version of Error Prone are you using?

Latest master

### Does this issue reproduce with the latest release?

Yes

### What did you do?

```java
import java.util.Locale;
class OrphanedTest {
  public static void main(String[] args) {
    StringBuilder b = new StringBuilder().append("Hello ").append("%s");
    String formatted = String.format(Locale.getDefault(), b.toString(), "world");
    System.out.println(formatted);
  }
}
```

### What did you expect to see?

No warning

### What did you see instead?

```
OrphanedTest.java:4: warning: [OrphanedFormatString] String literal contains format specifiers, but is not passed to a format method
    StringBuilder b = new StringBuilder().append("Hello ").append("%s");
                                                                  ^
    (see http://errorprone.info/bugpattern/OrphanedFormatString)
1 warning
```

**Comment by awturner** [comment]
Thanks for the report.

This will be fixed in the next release. The check on `StringBuilder.append` has been tightened to only consider invocations which pass parameters (i.e. `append(String, int, int)` and `append(char[], int, int)`).
